src/js/utils/MesosStateUtil.js

```diff
@@ -54,6 +54,41 @@ const MesosStateUtil = {
       });
   },
 
+  /**
+   * @param {Array.<Object>} resourceList - Verbose resource information
+   * @returns {Object} An object of only the resource values
+   */
+  extractExecutorResources(resourceList) {
+    const resources = {};
+
+    resourceList.forEach(resource => {
+      if (resource.type === "SCALAR") {
+        resources[resource.name] = resource.scalar.value;
+      }
+    });
+
+    return resources;
+  },
+
+  /**
+   * @param {Array} tasks
+   * @param {Array} executors
+   * @returns {Array} Tasks and executors concatenated
+   */
+  concatTasksExecutors(tasks, executors) {
+    const tasksExecutors = [].concat(tasks);
+
+    executors.forEach(executor =>
```

Comment:
🎨 Instead of using `forEach` and `push` here, you also could do `executors.map` to get a new array 🤔

Reply:
Much better! 💯